internal/providers/exoscale/exoscale.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2016 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// The Exoscale provider fetches a remote configuration from the
+// Exoscale user-data metadata service URL.
+
+package exoscale
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"time"
+
+	"github.com/coreos/ignition/v2/config/v3_1_experimental/types"
+	"github.com/coreos/ignition/v2/internal/providers/util"
+	"github.com/coreos/ignition/v2/internal/resource"
+
+	"github.com/coreos/vcontext/report"
+)
+
+var (
+	userdataURL = url.URL{
+		Scheme: "http",
+		Host:   "169.254.169.254",
+		Path:   "1.0/user-data",
+	}
+)
+
+func FetchConfig(f *resource.Fetcher) (types.Config, report.Report, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
```

Comment:
Hmm, I think we can just use `f.fetchToBuffer` directly here, right? See e.g. `aws.go`. IIRC, the client the fetcher uses already has a timeout (which can be overridden via the timeout Ignition config knob).

Reply:
Okay thanks, so I have simplified the code. I'll check timeout ignition knob. Have you got a link to that?